shared_model/utils/impl/amount_utils_impl.cpp

```diff
@@ -0,0 +1,121 @@
+/**
```

Comment:
Not really sure amount_utils_impl is a good name here. Amount utils does define any virtual methods